### DISCUSSION TITLE
HiGHS solver bindings

### DISCUSSION THREAD

**Discussion by dirkschumacher** [dirkschumacher/ompr#431] [General]
I create a small, experimental package to use HiGHS with ompr. Still a bit WIP, but already useable.

https://github.com/dirkschumacher/ompr.highs

HiGHS is a great open-source solver. More on HiGHS here: https://highs.dev

**Comment by datadrivensupplychain** [comment]
Would you recommend this package over `ROI.plugin.highs`? I was able to solve a toy model with `ROI.plugin.highs` but I couldn't get it to take even a MIPGap argument.

**Reply by dirkschumacher** [reply]
Since `ROI.plugin.highs` is on CRAN it may be better suited to depend on - depends on your use case. Not sure what arguments the ROI plugin registered. If you have a specific use case in mind maybe it's worth it to engage with the maintainer and see if you can contribute something.

**Comment by sbmack** [comment]
Dirk, thanks for the effort with HiGHS.  Unfortunately:

Installing from: https://github.com/dirkschumacher/ompr.highs

Throws a Warning:

* installing *source* package 'ompr.highs' ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
**Note: possible error in 'highs_solve(L = as.numeric(obj$solution), ': unused argument (dry_run = FALSE)** 
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (ompr.highs)

Executing the BIN Packing problem throws an error:
```
Error in highs_solve(L = as.numeric(obj$solution), A = constraints$matrix,  : 
  unused argument (dry_run = FALSE)
```

**Reply by datadrivensupplychain** [reply]
@sbmack I forked the repo, commented out ``dry_run=FALSE`` in the code, and installed from my github.  Works fine now. I also made a pull request to Dirk.
